packages/webapp/src/pages/Activity.tsx

```diff
@@ -216,11 +216,11 @@ export default function Activity() {
 
                 if (res?.status === 200) {
                     try {
-                        const allMessages = await res.json();
+                        const allMessages: ActivityMessageResponse = await res.json();
                         const logsWithMessages = activities.map((activity: ActivityResponse) => {
                             const logMessages = allMessages[activity.id];
                             if (logMessages) {
-                                activity.messages = logMessages;
+                                activity.messages = logMessages.reverse();
```

Comment:
can this be done in the query?

Reply:
I don't think it can be done in the query because of the `where`. If I change the order the `<=` will remove latest messages. 
e.g: 
```sh
# Right now the first messages are stripped down
[ { rank: 1, msg: last }, [...], { rank: 2000, msg: first }, ]
WHERE rank <= 1000

# If I change the order the last messages are stripped down
[ { rank: 1, msg: first }, [...], { rank: 2000, msg: last }, ]
WHERE rank <= 1000

# Maybe there is a way to do this. 
# But not sure if it would help the query plan because it would need to query everything instead of stopping at 1000.
[ { rank: 1, msg: first }, [...], { rank: 2000, msg: last }, ]
WHERE rank >= TOTAL - 1000
```

Reversing in the UI is just an easy way to off-load the reversing to the customer instead of doing this expensive operation in the code.